src/orion/benchmark/benchmark_client.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.benchmark.client` -- Benchmark client
+================================================================
+
+.. module:: client
+   :platform: Unix
+   :synopsis: Client to use Orion benchmark.
+
+"""
+import datetime
+
+from orion.benchmark import Benchmark, Study
+from orion.benchmark.assessment.base import BenchmarkAssessment
+from orion.benchmark.task.base import BenchmarkTask
+from orion.core.utils.exceptions import NoConfigurationError
+from orion.storage.base import get_storage, setup_storage
+
+
+def get_or_create_benchmark(
+    name, algorithms=None, targets=None, storage=None, debug=False
+):
+    """
+    Create or get a benchmark object.
+
+    Parameters
+    ----------
+    name: str
+        Name of the benchmark
+    algorithms: list, optional
+        Algorithms used for benchmark, each algorithm can be a string or dict.
+    targets: list, optional
+        Targets for the benchmark, each target will be a dict with two keys.
+
+        assess: list
+            Assessment objects
+        task: list
+            Task objects
+    storage: dict, optional
+        Configuration of the storage backend.
+    debug: bool, optional
+        If using in debug mode, the storage config is overrided with legacy:EphemeralDB.
+        Defaults to False.
+
+    Returns
+    -------
+    An instance of `orion.benchmark.Benchmark`
+    """
+    setup_storage(storage=storage, debug=debug)
+
+    # fetch benchmark from db
+    db_config = _fetch_benchmark(name)
+
+    benchmark_id = None
+
+    if db_config:
+        benchmark_id, algorithms, targets = _resolve_db_config(db_config)
+
+    if not algorithms or not targets:
+        raise NoConfigurationError(
+            "Benchmark {} does not exist in DB, "
+            "algorithms and targets space was not defined.".format(name)
+        )
+
+    benchmark = _create_benchmark(name, algorithms, targets)
+
+    if benchmark_id is None:
+        # persist benchmark into db
+        _register_benchmark(benchmark)
```

Comment:
In case of race conditions, this will raise a DuplicateKeyError and make the worker crash. One way to handle this is to retry recursively like we do here: https://github.com/Epistimio/orion/blob/develop/src/orion/core/io/experiment_builder.py#L187-L198.